File: 69sqrt(x)/sqrt(x).cpp
class Solution {
public:
    int mySqrt(int x) {
        auto [left, right]{std::pair{0, x / 2 + 1}};
        while (left != right)
        {
            const auto middle{left + (right - left) / 2};
            if (std::pow(middle, 2) < x) left = middle + 1;
            else right = middle;
        }
        return std::pow(left, 2) == x ? left : left - 1;
    }
};

File: 69sqrt(x)/sqrt(x).java
class Solution {
    public int mySqrt(int x) {
        int left = 0, right = x / 2 + 1;
        while (left != right)
        {
            final int middle = left + (right - left) / 2;
            if (Math.pow(middle, 2) < x) left = middle + 1;
            else right = middle;
        }
        return Math.pow(left, 2) == x ? left : left - 1;
    }
}
